### PR TITLE
🌱 Add cluster name on printer column on non core types

### LINKED ISSUE #5208: Add Cluster name as printer column on non core types
As per  https://github.com/kubernetes-sigs/cluster-api/pull/5202#issuecomment-912957671 we should add a Cluster column for  types in KCP, CABPK, extensions, CAPD.

Whenever the information is not available as explicit field, it should be possible to get this information from labels (see e.g. https://github.com/kubernetes-sigs/cluster-api/issues/5198#issuecomment-912717888 for an example of how this can be achived)

/kind feature

### PR DESCRIPTION
**Pull request by budhirajamadhav**
This PR will add clusterName as printer column on non core types

Fixes #5208

### CODE REVIEW

**Comment by budhirajamadhav** [comment]
Hey @fabriziopandini in the failed test of _pull-cluster-api-verify-main_ it says `generated files are out of date, run make generate`. Should I make a commit after performing `make generate`?

**Comment by JoelSpeed** [comment]
> Should I make a commit after performing make generate?

Yep that's correct, one you've run make generate there should be some differences to check in for the generated CRD manifests, these will need to be committed to fix the errors

File: bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go [hunk 1]
```diff
@@ -127,6 +127,7 @@ type KubeadmConfigStatus struct {
 // +kubebuilder:resource:path=kubeadmconfigs,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go:130]:
```suggestion
// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```

File: bootstrap/kubeadm/api/v1beta1/kubeadmconfigtemplate_types.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ type KubeadmConfigTemplateResource struct {
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:path=kubeadmconfigtemplates,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline bootstrap/kubeadm/api/v1beta1/kubeadmconfigtemplate_types.go:36]:
```suggestion
```
We don't set a cluster-name label on this resource

File: controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go [hunk 3]
```diff
@@ -201,6 +201,7 @@ type KubeadmControlPlaneStatus struct {
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
 // +kubebuilder:subresource:scale:specpath=.spec.replicas,statuspath=.status.replicas,selectorpath=.status.selector
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go:204]:
```suggestion
// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```

File: controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go [hunk 4]
```diff
@@ -28,6 +28,7 @@ type KubeadmControlPlaneTemplateSpec struct {
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:path=kubeadmcontrolplanetemplates,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go:31]:
```suggestion
```
We don't set a cluster-name label on this resource

File: test/infrastructure/docker/api/v1beta1/dockercluster_types.go [hunk 5]
```diff
@@ -97,6 +97,7 @@ type APIEndpoint struct {
 // +kubebuilder:subresource:status
 // +kubebuilder:storageversion
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline test/infrastructure/docker/api/v1beta1/dockercluster_types.go:100]:
```suggestion
// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```

File: test/infrastructure/docker/api/v1beta1/dockerclustertemplate_types.go [hunk 6]
```diff
@@ -27,6 +27,7 @@ type DockerClusterTemplateSpec struct {
 
 // +kubebuilder:object:root=true
 // +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline test/infrastructure/docker/api/v1beta1/dockerclustertemplate_types.go:30]:
```suggestion
```
We don't set a cluster-name label on this resource

File: test/infrastructure/docker/api/v1beta1/dockermachine_types.go [hunk 7]
```diff
@@ -94,6 +94,7 @@ type DockerMachineStatus struct {
 // +kubebuilder:object:root=true
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline test/infrastructure/docker/api/v1beta1/dockermachine_types.go:97]:
```suggestion
// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```

File: test/infrastructure/docker/api/v1beta1/dockermachinetemplate_types.go [hunk 8]
```diff
@@ -28,6 +28,7 @@ type DockerMachineTemplateSpec struct {
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:path=dockermachinetemplates,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline test/infrastructure/docker/api/v1beta1/dockermachinetemplate_types.go:31]:
```suggestion
```
We don't set a cluster-name label on this resource

File: test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_types.go [hunk 9]
```diff
@@ -114,6 +114,7 @@ type DockerMachinePoolInstanceStatus struct {
 // +kubebuilder:storageversion
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels['machine\\.cluster\\.x-k8s\\.io/cluster-name']",description="Cluster"
```
Comment (sbueringer) [inline test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_types.go:117]:
```suggestion
```
We don't set a cluster-name label on this resource

**Review by sbueringer — COMMENTED** [review]
Can you please make the suggested changes, then run `make generate` and `make -C test/infrastructure/docker generate-manifests` (the test failed because the second one was missing) and then squash it all into one commit? Thank you :)

And sorry there was a bit of context missing in the corresponding issue.

P.S. It would be great to get this PR into the 1.0 release (aka ASAP merged :))

File: controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go (hunk 4 above)
Comment (fabriziopandini) [reply controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go:31]: Agreed, this is a template introduced for ClusterClasses, so it is shared across many clusters.

File: test/infrastructure/docker/api/v1beta1/dockerclustertemplate_types.go (hunk 6 above)
Comment (fabriziopandini) [reply test/infrastructure/docker/api/v1beta1/dockerclustertemplate_types.go:30]: Agreed, this is a template introduced for ClusterClasses, so it is shared across many clusters.

File: test/infrastructure/docker/api/v1beta1/dockermachinetemplate_types.go (hunk 8 above)
Comment (fabriziopandini) [reply test/infrastructure/docker/api/v1beta1/dockermachinetemplate_types.go:31]: Agreed, also with ClusterClasses, templates are going to be shared across many clusters.

File: test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_types.go (hunk 9 above)
Comment (fabriziopandini) [reply test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_types.go:117]: Fine to remove the column for now but I consider this a bug, let's open an issue

Comment (sbueringer) [reply test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_types.go:117]: Agree

Comment (sbueringer) [reply test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_types.go:117]: Done > https://github.com/kubernetes-sigs/cluster-api/issues/5340

**Comment by fabriziopandini** [comment]
@budhirajamadhav if you can address comments we will try to get this into v1beta1 RC later today

**Comment by budhirajamadhav** [comment]
Yup @fabriziopandini I will do the required changes today itself. Out now, but will do it asap.

**Comment by budhirajamadhav** [comment]
Hey, @sbueringer if you're there was about to commit the changes after doing the suggested changes and performing `make generate`. But it edited so many files. Is it fine? Should I commit. Sorry if I'm asking a bit too much. It's just my first contribution.

image

**Comment by sbueringer** [comment]
@budhirajamadhav No problem. Looks like there's something wrong with deepcopy/conversion generation. I'm not sure what, but can you try reset and run `generate-manifests` instead of `generate` (in the main folder and in ./test/infrastructure/docker`)? This will not touch the deepcopy and conversion files.

Sorry I meant `generate-manifests` instead of `generate` (in my post above)

**Comment by budhirajamadhav** [comment]
Oho, no worries. I did make generate-manifests but I asked on slack and the thing was my go version was 1.17 and after performing `amke generate` it added "Go:" to the first line of every conversion/deepcopy file.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Comment by sbueringer** [comment]
@budhirajamadhav Thx :)

**Pull request merged**